src/clue.js

```diff
@@ -1,29 +1,190 @@
+      'She is from a wealthy family and uses her status and money to earn popularity',
+    image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+    color: 'blue'
+  },
+  {
+    firstName: 'Kasandra',
+    lastName: 'Scarlet',
+    occupation: 'Actor',
+    age: 31,
+    description: 'She is an A-list movie star with a dark past',
+    image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+    color: 'red'
+  },
+  {
+    firstName: 'Victor',
+    lastName: 'Plum',
+    occupation: 'Designer',
+    age: 22,
+    description: 'Billionaire video game designer',
+    image:
+      'https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg',
+    color: 'purple'
+  },
+  {
+    firstName: 'Doctor',
+    lastName: 'Orchid',
+    occupation: 'Scientist',
+    age: 26,
+    description: 'PhD in plant toxicology. Adopted daughter of Mr. Boddy',
+    image: 'http://www.radiotimes.com/uploads/images/Original/111967.jpg',
+    color: 'white'
+  },
+  {
+    firstName: 'Jacob',
+    lastName: 'Green',
+    occupation: 'Entrepreneur',
+    age: 45,
+    description: 'He has a lot of connections',
+    image:
+      'https://pbs.twimg.com/profile_images/506787499331428352/65jTv2uC.jpeg',
+    color: 'green'
+  }
+];
+
+// [mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard];
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = ['Dining Room', 'Conservatory', 'Kitchen', 'Study', 'Library', 'Billiard Room', 'Lounge', 'Ballroom', 'Hall', 'Spa', 'Living Room', 'Observatory', 'Theater', 'Guest House', 'Patio'];
 
 // Weapons Array
 
-const weaponsArray = [];
-
+const weaponsArray = [
+  {name: 'rope',
+weight: 10},
+{name:'knife',
+weight: 8},
+{name:'candlestick',
+weight: 2},
+{name:'dumbbell',
+weight: 30},
+{name:'poison',
+weight: 2},
+{name:'axe',
+weight: 15},
+{name:'bat',
+weight: 13},
+{name:'trophy',
+weight: 25},
+{name:'pistol',
+weight: 20}
+];
 
 // ITERATION 2
 
-function selectRandom() {}
 
-function pickMystery() {}
 
+function selectRandom(array) {
+  const randomSuspect = array[Math.floor(Math.random() * array.length)];
+  return randomSuspect;
+}
 
-// ITERATION 3
+function pickMystery() {
+const randomMystery ={};
+const randomSuspect = suspectsArray[Math.floor(Math.random() * suspectsArray.length)];
+randomMystery.suspect = randomSuspect;
+const randomRoom = roomsArray[Math.floor(Math.random() * roomsArray.length)];
+randomMystery.room = randomRoom;
+const randomWeapon = weaponsArray[Math.floor(Math.random() * weaponsArray.length)];
+randomMystery.weapon = randomWeapon;
+return randomMystery;
+};
 
-function revealMystery() {}
 
+// ITERATION 3
+
+function revealMystery(randomMystery) {
+pickMystery();
+const solution ='';
+// solution += randomMystery.suspect.firstName + ' ' + randomMystery.suspect.lastName + ' killed Mr.Boddy using the ' +randomMystery.weapon.weapon +' in the ' + randomMystery.room;
+return solution;
+}
+revealMystery();
```

Comment:
Lines 182 and 187 are unnecessary..

To help you figure out why it is not desired result, console.log `randomMystery`. So for example you can see below, you should access weapon by saying `randomMystery.weapon.name`:

```
{
  suspect: { firstName: 'aa', lastName: 'abc' },
  weapon: { name: 'abd' },
  room: { name: 'abb' }
}
```

And why test is not passing at the ned is the silliest thing! In the task , string to match has white space between` Mr.` and `Boddy`
`<FIRST NAME> <LAST NAME> killed Mr. Boddy using the <WEAPON> in the <ROOM>!`